src/adapters/mc/asb/CMakeLists.txt

```diff
@@ -19,41 +19,8 @@ target_link_libraries(OsConfigResourceAsb
-
-# For Audit (Get):
-
-add_custom_target(stage_create_asb_zip
-    COMMAND ${CMAKE_COMMAND} -E make_directory ${PROJECT_BINARY_DIR}/StagingAzureLinuxBaseline
-    COMMAND ${CMAKE_COMMAND} -E copy_if_different "${CMAKE_CURRENT_SOURCE_DIR}/AzureLinuxBaseline.metaconfig.json" ${PROJECT_BINARY_DIR}/StagingAzureLinuxBaseline/
-    COMMAND ${CMAKE_COMMAND} -E remove -f ${PROJECT_BINARY_DIR}/StagingAzureLinuxBaseline/SetAzureLinuxBaseline.metaconfig.json
-    COMMAND ${CMAKE_COMMAND} -E copy_if_different "${CMAKE_CURRENT_SOURCE_DIR}/AzureLinuxBaseline.mof" ${PROJECT_BINARY_DIR}/StagingAzureLinuxBaseline/
-    COMMAND ${CMAKE_COMMAND} -E remove -f ${PROJECT_BINARY_DIR}/StagingAzureLinuxBaseline/SetAzureLinuxBaseline.mof
-    COMMAND ${CMAKE_COMMAND} -E copy_if_different $<TARGET_FILE:OsConfigResourceAsb> ${PROJECT_BINARY_DIR}/StagingAzureLinuxBaseline/Modules/DscNativeResources/OsConfigResource/libOsConfigResource.so
-    DEPENDS OsConfigResourceAsb)
-
-add_custom_target(create_asb_zip ALL
-    BYPRODUCTS ${OsConfigRootBinaryDir}/AzureLinuxBaseline.zip
-    COMMAND ${CMAKE_COMMAND} -E tar "cfv" "${OsConfigRootBinaryDir}/AzureLinuxBaseline.zip" --format=zip .
-    DEPENDS stage_create_asb_zip
-    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/StagingAzureLinuxBaseline/)
-
-# For Remediation (Set):
-
-add_custom_target(stage_create_asb_zip_set
-    COMMAND ${CMAKE_COMMAND} -E make_directory ${PROJECT_BINARY_DIR}/StagingSetAzureLinuxBaseline
-    COMMAND ${CMAKE_COMMAND} -E copy_if_different "${CMAKE_CURRENT_SOURCE_DIR}/AzureLinuxBaseline.metaconfig.json" ${PROJECT_BINARY_DIR}/StagingSetAzureLinuxBaseline/SetAzureLinuxBaseline.metaconfig.json
-    COMMAND ${CMAKE_COMMAND} -E remove -f ${PROJECT_BINARY_DIR}/StagingSetAzureLinuxBaseline/AzureLinuxBaseline.metaconfig.json
-    COMMAND ${CMAKE_COMMAND} -E copy_if_different "${CMAKE_CURRENT_SOURCE_DIR}/AzureLinuxBaseline.mof" ${PROJECT_BINARY_DIR}/StagingSetAzureLinuxBaseline/SetAzureLinuxBaseline.mof
-    COMMAND ${CMAKE_COMMAND} -E remove -f ${PROJECT_BINARY_DIR}/StagingSetAzureLinuxBaseline/AzureLinuxBaseline.mof
-    COMMAND ${CMAKE_COMMAND} -E copy_if_different $<TARGET_FILE:OsConfigResourceAsb> ${PROJECT_BINARY_DIR}/StagingSetAzureLinuxBaseline/Modules/DscNativeResources/OsConfigResource/libOsConfigResource.so
-    DEPENDS OsConfigResourceAsb)
-
-add_custom_target(create_asb_zip_set ALL
-    BYPRODUCTS ${OsConfigRootBinaryDir}/SetAzureLinuxBaseline.zip
-    COMMAND ${CMAKE_COMMAND} -E tar "cfv" "${OsConfigRootBinaryDir}/SetAzureLinuxBaseline.zip" --format=zip .
-    DEPENDS stage_create_asb_zip_set
-    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/StagingSetAzureLinuxBaseline/)
+add_mc_package(
+    TARGET_NAME OsConfigResourceAsb
+    MOF_FILE ${CMAKE_CURRENT_SOURCE_DIR}/AzureLinuxBaseline.mof
+    META_CONFIG_JSON_FILE ${CMAKE_CURRENT_SOURCE_DIR}/AzureLinuxBaseline.metaconfig.json
+    DESTINATION_FILE AzureLinuxBaseline)
```

Comment:
Suggest to put back what it was as it was clearer and simpler. 

Also, please split this into a separate PR, has nothing to do with Telemetry,